quickstart/kubernetes-guestbook.md

```diff
@@ -0,0 +1,373 @@
+---
+title: "Create a Guestbook with Redis and PHP"
+---
+
+In this tutorial, we'll build and deploy
+[the standard Kubernetes Guestbook example](https://github.com/kubernetes/examples/tree/master/guestbook) using Pulumi.
+
+The guestbook is a simple, multi-tier web application that uses Redis and Nginx, powered by Docker containers and
+Kubernetes. The primary difference between this example and the standard Kubernetes one is that we'll be authoring it in
+TypeScript instead of YAML, and we'll deploy it using `pulumi` rather than `kubectl`. This gives us the full power of
+real languages, combined with immutable infrastructure with diffable updates, alongside reliable orchestration, status,
+and robust state management.
+
+The code for this tutorial is
+[available on GitHub](https://github.com/pulumi/examples/tree/master/kubernetes-ts-guestbook).
+
+## Objectives
+
+* Start up a Redis master
+* Start up Redis slaves
+* Start up the guestbook frontend
+* Expose and view the frontend service
+* Clean up
+
+## Before you begin
+
+You need to have the Pulumi CLI and a working Kubernetes cluster.
+[Minikube](https://kubernetes.io/docs/getting-started-guides/minikube) is an easy way to get started.
+
+1. [Install Pulumi](../install)
+2. [Connect Pulumi to a Kubernetes Cluster](../install/kubernetes.html)
+
+## Create a Project and Deploy It
+
+The guestbook application uses Redis to store its data. It writes its data to a Redis master instance and reads data
+from multiple Redis slave instances.
+
+Normally we would write YAML files to configure them, and then run `kubectl` commands to create and manage the services.
+Instead of doing that, we will author our program in code and deploy it with `pulumi`.
+
+To start, we'll need to create a project and stack (a deployment target) for our new project:
+
+1.  To create a new Pulumi project, let's use a template:
+
+    ```shell
+    $ pulumi new kubernetes-typescript --dir guestbook && cd guestbook
+    ```
+
+    There are a variety of ways to create a new project -- for instance, we could have started by cloning
+    [the above repo](https://github.com/pulumi/examples/tree/master/kubernetes-ts-guestbook) -- but templates are easy
+    and allow us walk through the process of creating a new project from scratch. This command will have initialized
+    a fresh project in the `guestbook` directory, and our command has `cd`'d into it.
+
+2.  Next, replace the minimal contents of the template's `index.js` file with the full guestbook:
```

Comment:
Is this TypeScript or JavaScript (the template above says `typescript`, this says `.js`, the code uses TypeScript features and is in a `js` code block)?

Reply:
I switched halfway through to TypeScript, since the example I'm basing it on and our current Kubernetes template is in TypeScript. I realize we tend to be JavaScript-first, although now that we have the first class TypeScript provider, I suspect it matters less (no manual `tsc` steps).

Reply:
And, of course, I will fix up to be consistent -- good catch.